diff --git a/learn/array_reduction_with.sv b/learn/array_reduction_with.sv
--- a/learn/array_reduction_with.sv	2026-08-16 23:31:57
+++ b/learn/array_reduction_with.sv	2026-08-16 23:31:57
@@ -23,7 +23,8 @@
 // -------------------------------------------------------------------------------------------------
 class bucket_item;
   rand bit [1:0] levels[8];
-  // write constraints here
+  constraint c_threes {levels.sum() with (item == 3 ? 1 : 0) == 2;}
+  constraint c_zeros {levels.sum() with (item == 0 ? 1 : 0) == 3;}
 endclass
 
 // ---8<--- checker below: don't edit ---
