diff --git a/learn/inherited_constraints_stack.sv b/learn/inherited_constraints_stack.sv
--- a/learn/inherited_constraints_stack.sv	2026-08-17 17:48:28
+++ b/learn/inherited_constraints_stack.sv	2026-08-17 17:48:28
@@ -35,7 +35,12 @@
 endclass
 
 class wide_item extends base_item;
-  // write constraints here
+  constraint c_wide {value inside {[0 : 200]};}
+
+  function new();
+    super.new();
+    c_range.constraint_mode(0);
+  endfunction
 endclass
 
 // ---8<--- checker below: don't edit ---
