diff --git a/exercises/sv/nonoverlapping_address_ranges.sv b/exercises/sv/nonoverlapping_address_ranges.sv
--- a/exercises/sv/nonoverlapping_address_ranges.sv
+++ b/exercises/sv/nonoverlapping_address_ranges.sv
@@ -28,8 +28,8 @@
     base[i] inside {[0 : 1000]};
   }
   constraint c_order {
-    base[0] < base[1];
-    base[1] < base[2];
+    base[0] + size[0] <= base[1];
+    base[1] + size[1] <= base[2];
   }
 endclass
 
