diff --git a/exercises/sv/pre_randomize_local_state.sv b/exercises/sv/pre_randomize_local_state.sv
--- a/exercises/sv/pre_randomize_local_state.sv
+++ b/exercises/sv/pre_randomize_local_state.sv
@@ -20,9 +20,9 @@
 class packet_item;
   rand bit [7:0] len;
   bit is_jumbo;
+  bit toggle;
 
   function void pre_randomize();
-    bit toggle = 0;
     toggle   = !toggle;
     is_jumbo = toggle;
   endfunction
