diff --git a/exercises/sv/risc_v_data_hazard.sv b/exercises/sv/risc_v_data_hazard.sv
--- a/exercises/sv/risc_v_data_hazard.sv
+++ b/exercises/sv/risc_v_data_hazard.sv
@@ -18,6 +18,22 @@
   rand bit [4:0] src_reg_0;
   rand bit [4:0] src_reg_1;
   rand bit       hazard;
+
+  bit      [4:0] prev_dst_reg;
+  bit            has_prev;
+
+  constraint c_hazard_dist {
+    hazard dist {
+      1 := 70,
+      0 := 30
+    };
+  }
+  constraint c_hazard_link {(has_prev && hazard) -> src_reg_0 == prev_dst_reg;}
+
+  function void post_randomize();
+    prev_dst_reg = dst_reg;
+    has_prev     = 1;
+  endfunction
 endclass
 
 // ---8<--- checker below: don't edit ---
