diff --git a/learn/uninitialized_dynamic_array.sv b/learn/uninitialized_dynamic_array.sv
--- a/learn/uninitialized_dynamic_array.sv
+++ b/learn/uninitialized_dynamic_array.sv
@@ -18,6 +18,7 @@
 // -------------------------------------------------------------------------------------------------
 class buf_item;
   rand int data[];
+  constraint c_size {data.size() inside {[1 : 8]};}
 endclass
 
 // ---8<--- checker below: don't edit ---
